Sprint_6

ncherny opened the pull request.

ncherny committed:
Added tests for Lion, Feline, Cat and used dependency injection for Feline inside Lion class

ncherny committed:
add jacoco report

ncherny committed:
Update README.md

ncherny committed:
Added encoding in maven plugin

okryukova commented on src/main/java/com/example/Lion.java:
```diff
@@ -6,18 +6,20 @@ public class Lion {
-    public Lion(String sex) throws Exception {
+    FelineInterface feline;
+
+    public Lion(String sex, FelineInterface felineInterface) throws Exception {
```
⛔️Нужно исправить.  Писать новый интерфейс не нужно. Используй Feline feline в параметрах конструктора

okryukova commented on src/test/java/CatTests.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Test
+    public void checkGetFoodReturnsValueFromPredator() {
+        List<String> expectedFood = List.of("Мясо", "Рыба");
```
⛔️Нужно исправить.  Мок должен вести себя как реальный объект и возвращать List.of("Животные", "Птицы", "Рыба");. 

Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/FelineTests.java:
```diff
@@ -0,0 +1,38 @@
+public class FelineTests {
+
+    @Spy
+    Feline feline = new Feline();
```
⛔️Нужно исправить.  Создавай моки только для зависимостей, а не для тестируемого объекта.

okryukova commented on src/test/java/LionCallsToFelineTests.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    public void checkGetKittensReturnsValueFromFeline() {
```
⚠️Можно улучшить. В названии метода необходимо использовать слово Test. Проверь, соблюдается ли данное правило для всех созданных тестов.

okryukova commented on src/test/java/LionManeSetupTests.java:
```diff
@@ -0,0 +1,38 @@
+    @ParameterizedTest(name = "{index}. Test data: Lion sex={0}")
+    @ValueSource(strings = {"", "СамкаСамец"})
+    public void checkManeSetupNegative(String lionSex) {
+        try {
```
⚠️Можно улучшить. В Junit4 есть метод assertThrows и @test(expected = Exception.class) для проверки исключений, try-cath нелучшая практика в тестах.

ncherny committed:
Fixed comments

- Removed Feline interface
- Fixed data for mock in Cat tests to use actual data
- Removed use of spy for testing of Feline in Feline tests
- Edited test method names to include "test"
- In LionManeSetupTests changed checking of exceptions to use
  assertThrows and assertDoesNotThrow
